runtime/centrifuge/src/migrations.rs

```diff
@@ -195,404 +208,316 @@ mod asset_registry {
-				get_centrifuge_assets()
-			} else {
-				get_catalyst_assets()
-			};
-
-			assets_to_migrate
-				.iter()
-				.for_each(|(asset_id, asset_metadata)| {
-					orml_asset_registry::Pallet::<T>::do_register_asset_without_asset_processor(
-						(*asset_metadata).clone().into(),
-						(*asset_id).into(),
-					)
-					.map_err(|e| log::error!("Failed to register asset: {:?}", e))
-					.ok();
-				});
-
-			loc_count = orml_asset_registry::LocationToAssetId::<T>::iter_keys().count() as u32;
-			meta_count = orml_asset_registry::Metadata::<T>::iter_keys().count() as u32;
-
-			log::info!("After Found {} LocationToAssetId keys ", loc_count);
-			log::info!("After Found {} Metadata keys ", meta_count);
-
-			log::info!("AssetRegistryMultilocationToXCMV3: on_runtime_upgrade: completed!");
-			RocksDbWeight::get().reads_writes((meta_count * 2) as u64, (2 + meta_count) as u64)
 		}
+	}
 
-		#[cfg(feature = "try-runtime")]
-		fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
-			use frame_support::storage::StoragePrefixedMap;
-
-			let loc_module_prefix = orml_asset_registry::LocationToAssetId::<T>::module_prefix();
-			let loc_storage_prefix = orml_asset_registry::LocationToAssetId::<T>::storage_prefix();
-			let loc_target_prefix =
-				frame_support::storage::storage_prefix(loc_module_prefix, loc_storage_prefix);
-
-			let meta_module_prefix = orml_asset_registry::Metadata::<T>::module_prefix();
-			let meta_storage_prefix = orml_asset_registry::Metadata::<T>::storage_prefix();
-			let meta_target_prefix =
-				frame_support::storage::storage_prefix(meta_module_prefix, meta_storage_prefix);
-
-			let loc_count = count_storage_keys(&loc_target_prefix);
-			let meta_count = count_storage_keys(&meta_target_prefix);
-
-			log::info!("Found {} LocationToAssetId keys ", loc_count);
-			log::info!("Found {} Metadata keys ", meta_count);
-
-			let is_centrifuge = meta_count == 6;
-
-			let mut expected_loc_count = 5;
-			let mut expected_meta_count = 6;
-			if !is_centrifuge {
-				expected_loc_count = 2;
-				expected_meta_count = 3;
-			}
-			frame_support::ensure!(
-				loc_count == expected_loc_count,
-				"Pre: LocationToAssetId Unexpected storage state"
-			);
-
-			frame_support::ensure!(
-				meta_count == expected_meta_count,
-				"Pre: Metadata Unexpected storage state"
-			);
-
-			Ok(Default::default())
+	impl CentrifugeAssets {
+		fn get_centrifuge_assets() -> Vec<(
+			CurrencyId,
+			orml_asset_registry::AssetMetadata<Balance, CustomMetadata>,
+		)> {
+			let mut gk = [0u8; 32];
+			gk[..2].copy_from_slice(b"01");
+
+			vec![
+				(
+					CurrencyId::Native,
+					orml_asset_registry::AssetMetadata {
+						decimals: 18,
+						name: b"Centrifuge".to_vec(),
+						symbol: b"CFG".to_vec(),
+						existential_deposit: 1_000_000_000_000u128,
+						location: Some(VersionedMultiLocation::V3(MultiLocation::new(
+							0,
+							Junctions::X1(GeneralKey {
+								length: 2,
+								data: gk,
+							}),
+						))),
```

Comment:
I think this must be 

```suggestion
						location: Some(VersionedMultiLocation::V3(MultiLocation::new(
							1,
							Junctions::X2(
							    Parachain(ParachainInfo::parachain_id().into()),
							    GeneralKey {
								length: 2,
								data: gk,
							}),
						))),
```

Reply:
cc @NunoAlexandre as I think in your PR you came to the conclusion that we can better go with the absolut locations in the registry.